plugins/pipelines/create.js

```diff
@@ -113,6 +120,36 @@ module.exports = () => ({
 
                                                 return defaultCollection.update();
                                             })
+                                            .then(() => {
+                                                if (autoKeysGeneration) {
```

Comment:
This boolean check if scm, such as `github` has ability to do `autoKeysGeneration`.

What if such scm, say `bitbucket` does not have such feature and `autoKeysGeneration = true` from the payload. How do we handle that case? Would there be: 

1) add a `catch` block
2) add another if condition to check both payload sent from ui has `autoKeysGeneration: true`, as well as check if given SCM has such ability.

In UI, we can prevent user giving input for if scm does not have `autoKeysGeneration` feature, but we also need to sanitize and guard on API. I can see some power users would call this API directly.


Just a thought, what do you think?


Note: 

From: https://github.com/screwdriver-cd/screwdriver/issues/1079#issuecomment-393631839

If we apply this feature to all SCMs: `scm-github`, `scm-bitbucket`, then we dont have such issue, just need to catch the errors.

Reply:
Yeah I get the point. I didn't take the case of direct API calls.
Catch is definitely needed. And it will be handled by the not implemented promise rejection in scm base.
I am wondering if the 2nd point is needed because if `autoKeysGeneration` is set to true in the payload, it will be rejected in the addDeployKey method and eventually sent to the catch block.

Reply:
Cool. I meant either 1) or 2). 

Since handled that case in 1), 2) is no longer necessary

Reply:
alright cool 👍

Reply:
don't we already have a `catch` at end of Promise chain ?